modules/images/webp-uploads/load.php

```diff
@@ -5,40 +5,294 @@
+ * for example an image/jpeg can be converted into an image/webp.
+ *
+ * @since n.e.x.t
+ *
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function webp_uploads_get_supported_image_mime_transforms() {
+	$image_mime_transforms = array(
+		'image/jpeg' => array( 'image/webp' ),
+		'image/webp' => array( 'image/jpeg' ),
+	);
 
-	return $output_format;
+	/**
+	 * Filter to allow the definition of a custom mime types, in which a defined mime type
+	 * can be transformed and provide a wide range of mime types.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, array<string>> An array with the valid mime transforms.
+	 *
+	 * @return array<string, array<string>> An array with the valid mime transformation
+	 */
+	return (array) apply_filters( 'webp_uploads_supported_image_mime_transforms', $image_mime_transforms );
+}
+
+/**
+ * Hook fired when an attachment is deleted, this hook is in charge of removing any
+ * additional mime types created by this plugin besides the original image. Any source
+ * with the same as the main image would not be removed by this hook due this file would
+ * be removed by WordPress when the attachment is deleted, usually this happens after this
+ * hook is executed.
+ *
+ * @since n.e.x.t
+ *
+ * @see   wp_delete_attachment
+ *
+ * @param int $attachment_id The ID of the attachment the sources are going to be deleted.
+ *
+ * @return void
+ */
+function webp_uploads_remove_sources_files( $attachment_id ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+
+	if (
+		! isset( $metadata['sizes'], $metadata['file'] )
+		|| empty( $metadata['file'] )
+		|| ! is_array( $metadata['sizes'] )
+	) {
+		return;
+	}
+
+	$upload_path = wp_get_upload_dir();
+	if ( empty( $upload_path['basedir'] ) ) {
+		return;
+	}
+
+	$file_directory = dirname( $metadata['file'] );
+	$directory      = path_join( $upload_path['basedir'], $file_directory );
+
+	foreach ( $metadata['sizes'] as $size ) {
+		if ( ! isset( $size['sources'] ) || ! is_array( $size['sources'] ) ) {
+			continue;
+		}
+
+		$original_size_mime = empty( $size['mime-type'] ) ? '' : $size['mime-type'];
+
+		foreach ( $size['sources'] as $mime => $properties ) {
+			/**
+			 * When we face the same mime type as the original image, we ignore this file as this file
+			 * would be removed when the size is removed by WordPress itself. The meta information as well
+			 * would be deleted as soon as the image is removed.
+			 *
+			 * @see wp_delete_attachment
+			 */
+			if ( $original_size_mime === $mime ) {
+				continue;
+			}
+
+			if ( ! is_array( $properties ) || empty( $properties['file'] ) ) {
+				continue;
+			}
+
+			$file_deletion = path_join( $file_directory, $properties['file'] );
+			$file_deletion = path_join( $upload_path['basedir'], $file_deletion );
+			wp_delete_file_from_directory( $file_deletion, $directory );
```

Comment:
See above - also, more importantly, the logic here looks different from WP core's - are you sure that works? Here's how it would be, following WP core's `wp_delete_attachment_files()`:
```suggestion
			$intermediate_file = str_replace( wp_basename( $metadata['file'] ), $properties['file'], $metadata['file'] );
			if ( ! empty( $intermediate_file ) ) {
				$intermediate_file = path_join( $uploadpath['basedir'], $intermediate_file );
				wp_delete_file_from_directory( $intermediate_file, $intermediate_dir );
			}
```

Reply:
The current logic works, but I can verify that particular case from core.